src/server/pfs/server/driver.go

```diff
@@ -1974,152 +1639,83 @@ func (d *driver) createBranch(txnCtx *txncontext.TransactionContext, branch *pfs
 	if len(provenance) > 0 && trigger != nil {
 		return errors.New("a branch cannot have both provenance and a trigger")
 	}
-
 	var err error
 	if err := d.env.AuthServer.CheckRepoIsAuthorizedInTransaction(txnCtx, branch.Repo, auth.Permission_REPO_CREATE_BRANCH); err != nil {
 		return errors.EnsureStack(err)
 	}
 	// Validate request
-	if err := ancestry.ValidateName(branch.Name); err != nil {
+	var allBranches []*pfs.Branch
+	allBranches = append(allBranches, provenance...)
+	allBranches = append(allBranches, branch)
+	for _, b := range allBranches {
+		if err := ancestry.ValidateName(b.Name); err != nil {
+			return err
+		}
+	}
+	// Create any of the provenance branches that don't exist yet, and give them an empty commit
```

Comment:
I think we should remove this functionality unless there is a reason why we absolutely need it. The user value doesn't outweigh the additional complexity imo.